### DISCUSSION TITLE
Finding grid files on NCAR or NERSC systems

### DISCUSSION THREAD

**Discussion by pgoddard08** [UXARRAY/uxarray#1111] [Q&A] [answered]
Hello,
I need to find the grid for E3SMv2 output. I see in "Plotting E3SMv2 Output with UXarray", this path is used: 
"grid_file = (
    "/glade/campaign/cisl/vast/uxarray/data/e3sm_keeling/E3SM_grid/ne30pg2_grd.nc"
)

I don't have permissions for this path. How can I get the ne30pg2_grd.nc file? I have access both to NCAR and NERSC systems.
Thanks!
Paul

**Comment by rajeeja** [comment]
Thanks for the  question. Where do you see it?

**Reply by xylar** [reply]
https://uxarray.readthedocs.io/en/latest/examples/visualization/e3sm-load-viz.html

**Reply by xylar** [reply]
Also:
https://uxarray.readthedocs.io/en/stable/examples/e3sm-calc-workflow.html

**Reply by xylar** [reply]
The files are presumably at NCAR somewhere and if one has an account, one can run the notebooks.  But I agree with @pgoddard08 that it would be nice to host the files somewhere public so someone without an NCAR account can also run the notebooks.

**Comment by philipc2** [comment]
Hi @pgoddard08

I can coordinate with @rytam2 to make these files readable on the the NCAR systems. 

Storing these publicly in the cloud somewhere as @xylar mentioned is something we've been wanted to do for a while now. If you have any suggestions, we'd be more than happy to look into it.

**Reply by xylar** [reply]
E3SM has a public data server and that seems like the easiest place to put these particular files. If someone puts them on a Google drive or lets me know who from E3SM supplied them in the first place, I can find an appropriate place to host them.

**Reply by xylar** [reply]
I tried to find this run in our documentation and didn't see is so I don't think I'll be able to track down these files myself.

**Reply by pgoddard08** [reply]
Great, thanks @philipc2 and @xylar . I really appreciate your help. Please let me know when these files are either readable on NCAR or housed somewhere for the general public. I've found such grids used for this example and used for general regridding difficult to find, so a public domain from the E3SM website would be ideal. Many thanks!

**Reply by philipc2** [reply]
@xylar 

I've stored the grid file & a selection of data variables in the following [google drive folder](https://drive.google.com/drive/folders/1RMmnvfRIvb8puK__XwzP9Oah88bYSu2D?usp=sharing).

I've included the data variables both in `zarr` and `netcdf` file formats. Feel free to use whichever format is better suited for the data server. 

Thanks again!

**Reply by xylar** [reply]
@philipc2, I'm happy to host both but wondering about the zarr file, which is actually a directory on GoogleDrive.  It seems completely unwieldy for folks to download an entire directory over https.  I don't have much experience with zarr format but is there a way to bundle it up more conveniently?

**Reply by xylar** [reply, marked as answer]
In any case, I have the data here:
https://web.lcrc.anl.gov/public/e3sm/diagnostics/uxarray_data/ENSO_ctl_1std/
If you want me to tar up the zarr data or something like that, let me know.  Feel free to point to that location in your documentation, etc.

**Reply by philipc2** [reply]
Hi @xylar 

Thank you so much for uploading the data! 

> It seems completely unwieldy for folks to download an entire directory over https. I don't have much experience with zarr format but is there a way to bundle it up more conveniently?

Zarr files are stored as a directory instead of as a single dataset. Since it is already compressed, I believe zipping it should also be an appropriate way to store it. 

Pinging @erogluorhan if he has anything to add.

**Reply by erogluorhan** [reply]
Hi @xylar , thanks a lot for your help with storing these on E3SM public data server!

Zarr should already be chunked (I don't know the details of how this zarr file was created though) and compressed. While it is optimized for object storage systems, it should also be suitable for being efficiently accessed via https without downloading the entire directory AFAIK (e.g. xarray's `open_dataset()` using Zarr as the engine). 

That said, if everything works as expected, hosting such a relatively large dataset in the zarr format than netcdf would be beneficial for future analysis workflows (I am not against keeping the netcdf file that you already put there though).

Please note @philipc2 : Some Zarr-based workflows mentioned metadata being stored in additional Zarr formats such as `.zgroup` etc. I can see the separate directories for several variables in this Zarr directory you created; does each of them have the metadata embedded into itself? Otherwise, should we have metadata explicitly created?

**Reply by xylar** [reply]
Okay, I'm in the process of adding a `E3SM_ENSO_ctl_keeling_eam_h0_0006-12.zarr.zip` file.

**Reply by xylar** [reply]
I'm also going to delete the zarr directory because I don't think it will be super useful and it's a bit of redundant storage.

**Reply by xylar** [reply]
Sorry, somehow didn't see your message @erogluorhan until just now.  I'll put back the `zarr` directory and delete the zip file then.

**Reply by xylar** [reply]
The metadata (`.zattrs`, `.zgroup` and `.zmetadata`) are there but don't seem visible on the web server.  Let me see if they are downloadable.

**Reply by xylar** [reply]
Yes, it looks like it:
https://web.lcrc.anl.gov/public/e3sm/diagnostics/uxarray_data/ENSO_ctl_1std/E3SM_ENSO_ctl_keeling_eam_h0_0006-12.zarr/.zmetadata

**Reply by erogluorhan** [reply]
Thanks a lot!
